sklearn/ensemble/tests/test_weight_boosting.py

```diff
@@ -505,6 +496,52 @@ def test_multidimensional_X():
     boost.predict(X)
 
 
+class ClassifierWithoutWeight(DummyClassifier):
+    """Classifier not supporting `sample_weight`."""
+    def fit(self, X, y):
+        return super().fit(X, y)  # pragma: no cover
+
+
+@pytest.mark.parametrize("algorithm", ['SAMME', 'SAMME.R'])
+def test_adaboostclassifier_without_sample_weight(algorithm):
+    X, y = iris.data, iris.target
+    base_estimator = ClassifierWithoutWeight()
+    clf = AdaBoostClassifier(
+        base_estimator=base_estimator, algorithm=algorithm
+    )
+    err_msg = ("{} doesn't support sample_weight"
+               .format(base_estimator.__class__.__name__))
+    with pytest.raises(ValueError, match=err_msg):
+        clf.fit(X, y)
+
+
+def test_adaboost_regressor_sample_weight():
+    # check that giving weight will have an influence on the error computed
+    # for a weak learner
+    X, y = datasets.make_regression(n_features=50, random_state=0)
+
+    # add an arbitrary outlier to make sure
+    X = np.vstack([X, X.sum(axis=0)])
+    y = np.hstack([y, 10])
+
+    regr_no_outlier = AdaBoostRegressor(
+        base_estimator=LinearRegression(), n_estimators=4, random_state=0
+    )
+    regr_with_weight = clone(regr_no_outlier)
+
+    # fit 2 models:
+    # - a model without the outlier
+    # - a model containing the outlier but with a null sample-weight
+    # Therefore, the error of the first weak learner will be identical.
+    regr_no_outlier.fit(X[:-1], y[:-1])
+    sample_weight = np.array([1.] * (y.size - 1) + [0.])
```

Comment:
I find it more clear if you do
```
sample_weight = np.ones(y.size)
sample_weight[-1] = 0
```
to explicitly show you're putting a 0 weight on the outlier